content/en/blog/_posts/2023-03-22-recent-developments-in-kubelet-to-speed-up-pod-startup.md

```diff
@@ -0,0 +1,56 @@
+
+In versions prior to v1.27, the default `kubeAPIQPS` was 5 and `kubeAPIBurst` was 10. However, Kubelet has increased these defaults to 50 and 100 respectively for better performance during pod startup. It's worth noting that this isn't the only reason why we've bumped up the API QPS limits for Kubelet.
+
+1. It has a potential to be hugely throttled now (default QPS = 5)
+2. In large clusters they can generate significant load anyway as there are a lot of them
+3. They have a dedicated PriorityLevel and FlowSchema that we can easily control
+
+In old days, we always encounter `volume mount timeout` on kubelet in node with more than 50 pods during pod start up. We suggest our customers to bump `kubeAPIQPS` to 20 and `kubeAPIBurst` to 40 and especially with customers that is using bare-metal(physical machines).
+
+More detials can be found in the KEP <https://kep.k8s.io/1040> and <https://github.com/kubernetes/kubernetes/pull/116121>.
+
+## Evented PLEG will enhance the scalability of kubelet
+
+`Evented PLEG` is set to be in beta for v1.27, disabled by default (needs to be enabled explicitly). It aids pod startup by reducing unnecessary work during inactivity through replacing periodic polling. The current `Generic PLEG` incurs non-negligible overhead due to frequent container status polling, which worsens with Kubelet's parallelism, leading to poor performance and reliability issues. For these reasons, it is recommended that we use Evented PLGE instead. Further details can be found in the KEP <https://kep.k8s.io/3386>.
+
+## What's more?
+
+In Kubernetes v1.26, a new histogram metric `pod_start_sli_duration_seconds` was added for Pod startup latency SLI/SLO details. Additionally, the kubelet log will now display more information about pod start-related timestamps, as shown below:
+
+> Dec 30 15:33:13.375379 e2e-022435249c-674b9-minion-group-gdj4 kubelet[8362]: I1230 15:33:13.375359    8362 pod_startup_latency_tracker.go:102] "Observed pod startup duration" pod="kube-system/konnectivity-agent-gnc9k" podStartSLOduration=-9.223372029479458e+09 pod.CreationTimestamp="2022-12-30 15:33:06 +0000 UTC" firstStartedPulling="2022-12-30 15:33:09.258791695 +0000 UTC m=+13.029631711" lastFinishedPulling="0001-01-01 00:00:00 +0000 UTC" observedRunningTime="2022-12-30 15:33:13.375009262 +0000 UTC m=+17.145849275" watchObservedRunningTime="2022-12-30 15:33:13.375317944 +0000 UTC m=+17.146157970"
+
+To identify the cause of slow pod startup, analyzing metrics and logs can be helpful. Other factors that may impact pod startup include container runtime, disk speed, CPU and memory resources on the node.
+
+SIG Node is responsible for ensuring fast Pod startup times, while addressing issues in large clusters falls under the purview of SIG Scalability as well.
```

Comment:
This doesn't seem to link to the previous text. Is it needed?

Reply:
I wanted users can know which SIGs care more about the pod startup speed.